packages/launch/src/hooks/use-domain-selection.ts

```diff
@@ -42,17 +43,36 @@ export function useDomainSuggestionFromCart(): DomainSuggestions.DomainSuggestio
 
 	const domainName = domainProductFromCart?.meta;
 
-	const domainSuggestion = useSelect( ( select ) =>
-		domainName
-			? select( DOMAIN_SUGGESTIONS_STORE ).getDomainSuggestions( domainName, {
-					quantity: 1,
-					include_wordpressdotcom: false,
-					include_dotblogsubdomain: false,
-			  } )?.[ 0 ]
-			: undefined
+	const domainDetails = useSelect( ( select ) =>
+		select( DOMAIN_SUGGESTIONS_STORE ).isAvailable( domainName || '' )
 	);
 
-	return domainSuggestion;
+	// if the domain is still available, forge a domain suggestion from it and return it
+	if (
+		domainProductFromCart &&
+		domainName &&
+		domainDetails &&
+		// for now, we consider premium domains unavailable and don't convert them into suggestions
+		// [ 'available', 'available_premium' ].indexOf( domainDetails.status ) > -1
```

Comment:
I don't understand this comment, could you explain it to me again? Thank you!

Reply:
Reworded the comment.